ui/raidboss/emulator/data/PopupTextAnalysis.ts

```diff
@@ -26,70 +68,71 @@ class Resolver {
       this.final();
     return true;
   }
-  setDelay(delayUntil) {
+  setDelay(delayUntil: number): Promise<void> {
     this.delayUntil = delayUntil;
     return this.delayPromise = new Promise((res) => {
       this.delayResolver = res;
     });
   }
-  setPromise(promise) {
+  setPromise(promise: Promise<void>) {
     this.promise = promise;
   }
-  setRun(run) {
+  setRun(run: ResolverFunc) {
     this.run = run;
   }
-  setFinal(final) {
+  setFinal(final: ResolverFunc) {
     this.final = final;
   }
-  setHelper(triggerHelper) {
+  setHelper(triggerHelper: EmulatorTriggerHelper) {
     this.triggerHelper = triggerHelper;
   }
 }
 
 export default class PopupTextAnalysis extends StubbedPopupText {
-  constructor(options, timelineLoader, raidbossFileData) {
-    super(options, timelineLoader, raidbossFileData);
-    this.triggerResolvers = [];
-  }
+  triggerResolvers: Resolver[] = [];
+  currentResolver?: Resolver;
+  public callback?: (log: LineEvent,
+      triggerHelper: EmulatorTriggerHelper | undefined,
+      currentTriggerStatus: ResolverStatus,
+      finalData: DataType) => void;
 
   // Override `OnTrigger` so we can use our own exception handler
-  OnTrigger(trigger, matches, currentTime) {
+  OnTrigger(trigger: LooseTrigger, matches: RegExpExecArray | null, currentTime: number): void {
     try {
       this.OnTriggerInternal(trigger, matches, currentTime);
     } catch (e) {
       console.log(trigger, e);
     }
   }
 
-  async OnLog(e) {
+  async OnLog(e: EmulatorLogEvent): Promise<void> {
     for (const logObj of e.detail.logs) {
-      const log = logObj.properCaseConvertedLine || logObj.convertedLine;
+      const log = logObj.properCaseConvertedLine ?? logObj.convertedLine;
 
       if (log.includes('00:0038:cactbot wipe'))
         this.SetInCombat(false);
 
       for (const trigger of this.triggers) {
-        const r = log.match(trigger.localRegex);
+        // partial commit, make sure you amend!
```

Comment:
?

Reply:
Ah, I left that note to myself when changing branches, came back, amended the commit to finish the conversion, and forgot to remove that note. 😊